Tests/Common.ps1

```diff
@@ -33,29 +33,38 @@ function Initialize-CommonTestSetup
     param()
 
     $moduleRootPath = Split-Path -Path $PSScriptRoot -Parent
-    . (Join-Path -Path $moduleRootPath -ChildPath 'Tests\Config\Settings.ps1')
+    $settingsPath = Join-Path -Path $moduleRootPath -ChildPath 'Tests/Config/Settings.ps1'
+    . $settingsPath
     Import-Module -Name (Join-Path -Path $moduleRootPath -ChildPath 'PowerShellForGitHub.psd1') -Force
 
+    $originalSettingsHash = (Get-GitHubConfiguration -Name TestConfigSettingsHash)
+    $currentSettingsHash = Get-SHA512Hash -PlainText (Get-Content -Path $settingsPath -Raw -Encoding Utf8)
+    $settingsAreUnaltered = $originalSettingsHash -eq $currentSettingsHash
+
     if ([string]::IsNullOrEmpty($env:ciAccessToken))
     {
-        $message = @(
-            'The tests are using the configuration settings defined in Tests\Config\Settings.ps1.',
-            'If you haven''t locally modified those values, your tests are going to fail since you',
-            'don''t have access to the default accounts referenced.  If that is the case, you should',
-            'cancel the existing tests, modify the values to ones you have access to, call',
-            'Set-GitHubAuthentication to cache your AccessToken, and then try running the tests again.')
-        Write-Warning -Message ($message -join [Environment]::NewLine)
+        if ($settingsAreUnaltered) {
+            $message = @(
+                'The tests are using the configuration settings defined in Tests/Config/Settings.ps1.',
+                'If you haven''t locally modified those values, your tests are going to fail since you',
+                'don''t have access to the default accounts referenced.  If that is the case, you should',
+                'cancel the existing tests, modify the values to ones you have access to, call',
+                'Set-GitHubAuthentication to cache your AccessToken, and then try running the tests again.')
+            Write-Warning -Message ($message -join [Environment]::NewLine)
+        }
     }
     else
     {
-        $secureString = $env:ciAccessToken | ConvertTo-SecureString -AsPlainText -Force
-        $cred = New-Object System.Management.Automation.PSCredential "<username is ignored>", $secureString
-        Set-GitHubAuthentication -Credential $cred
+        if(-not [string]::IsNullOrEmpty($env:ciAccessToken)) {
```

Comment:
It's not clear to me why you have this condition.  We're already in this state as a result of being in the `else` of the outer `if` that checked `$env:ciAccessToken` already...

Reply:
Thanks!